### PR TITLE
feat: Added --with-terraform argument to install coder and terraform together

### PR DESCRIPTION
**Pull request by nanospearing**
- Added a --install-terraform argument
- Added a unzip command check to the standalone function
- Cleaner error and help redirect the user to a solution
- Added help info for --with-terraform argument
- Fixed standalone install typo (ard64 -> arm64)

I have read the CLA Document and I hereby sign the CLA

### CODE REVIEW

**Comment by KozmikNano** [comment]
I have read the CLA Document and I hereby sign the CLA

File: install.sh [hunk 1]
```diff
@@ -385,6 +440,13 @@ install_apk() {
 }
 
 install_standalone() {
+	# Check if the unzip package is installed. If not error peacefully.
+	if !(command_exists unzip); then
+		echoh
+		echoerr "This script needs the unzip package to run."
+		echoerr "Please install unzip to use this function"
+		exit 1
+	fi
```
Comment (kylecarbs) [inline install.sh:449]: I don't believe unzip is needed on all platforms, right?

**Comment by KozmikNano** [comment]
recheck

File: install.sh (hunk 1 above)
Comment (kylecarbs) [reply install.sh:449]: Only darwin needs it:
image

File: install.sh [hunk 2]
```diff
@@ -351,6 +369,43 @@ fetch() {
...
 
+install_terraform() {
+	# Check if the unzip package is installed. If not error peacefully.
+	if !(command_exists unzip); then
+		echoh
+		echoerr "This script needs the unzip package to run."
+		echoerr "Please install unzip to use this function"
+		exit 1
+	fi
+	TERRAFORM_VERSION="1.3.4"
```
Comment (kylecarbs) [inline install.sh:380]: Could we move this higher up so the URL above auto-includes this too?

File: install.sh [hunk 3]
```diff
@@ -351,6 +369,43 @@ fetch() {
...
+	echoh "Installing Terraform version $TERRAFORM_VERSION $TERRAFORM_ARCH package from Hashicorp Source."
```
Comment (kylecarbs) [inline install.sh:381]:
```suggestion
	echoh "Installing Terraform version $TERRAFORM_VERSION $TERRAFORM_ARCH from the HashiCorp release repository."
```

File: install.sh [hunk 4]
```diff
@@ -212,6 +221,9 @@ main() {
 			usage
 			exit 0
 			;;
+		--with-terraform)
+			METHOD=install_terraform
```
Comment (kylecarbs) [inline install.sh:225]: Should we call this `with_terraform` too for consistency?

**Review by kylecarbs — COMMENTED** [review]
Minor nits, overall awesome contribution!

p.s. thanks for fixing the `ard64` -> `arm64` 🙈

File: install.sh (hunk 1 above)
Comment (KozmikNano) [reply install.sh:449]: I can remove that part if not using darwin in the standalone function, but for the terraform install section it is needed. Some linux distros, somehow, don't have it preinstalled and it is needed as the terraform binaries come in zip files.

File: install.sh (hunk 2 above)
Comment (KozmikNano) [reply install.sh:380]: Yes, I will do that real quick.

File: install.sh (hunk 1 above)
Comment (kylecarbs) [reply install.sh:449]: Yup, makes sense.

File: install.sh (hunk 2 above)
Comment (KozmikNano) [reply install.sh:380]: How do you think putting it on line 160 would feel?
```
main() {
	if [ "${TRACE-}" ]; then
		set -x
	fi
	TERRAFORM_VERSION="1.3.4"
	unset \
		DRY_RUN \
		METHOD \
		OPTIONAL \
		ALL_FLAGS \
		RSH_ARGS \
		EDGE \
		RSH

	ALL_FLAGS=""
	while [ "$#" -gt 0 ]; do
		case "$1" in
		-*)
			ALL_FLAGS="${ALL_FLAGS} $1"
			;;
		esac
...
```

File: install.sh (hunk 4 above)
Comment (KozmikNano) [reply install.sh:225]: Sure!

**Comment by kylecarbs** [comment]
@KozmikNano could you run fmt and fix the shellcheck?

**Comment by KozmikNano** [comment]
> @KozmikNano could you run fmt and fix the shellcheck?

Yes! Sorry for the late reply.

**Comment by KozmikNano** [comment]
@kylecarbs Should be good now.

File: install.sh (hunk 2 above)
Comment (KozmikNano) [reply install.sh]: Will putting it above the `if` statement be good, or higher?

Comment (kylecarbs) [reply install.sh]: Seems fine!

Comment (KozmikNano) [reply install.sh]: Cool, I will add the commit real quick!

**Comment by KozmikNano** [comment]
Thank you for your continued patience!

**Comment by kylecarbs** [comment]
@KozmikNano thank you for the contribution! Much appreciated.

**Comment by KozmikNano** [comment]
> @KozmikNano thank you for the contribution! Much appreciated.

Anytime, glad I could help!

**Comment by matifali** [comment]
Did you also update the related docs and main readme? @KozmikNano

**Comment by KozmikNano** [comment]
> Did you also update the related docs and main readme? @KozmikNano

I didn't see anything in the docs that referenced the command list, however I did add information to help argument in install.sh.

**Review by kylecarbs — APPROVED** [review]

**Pull request merged**